### PR TITLE
Non blocking/8 sign in

### PR DESCRIPTION
**Pull request by wangmingliang-ms**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java [hunk 1]
```diff
@@ -176,8 +179,16 @@ protected JComponent createCenterPanel() {
...
+    public Single<AuthMethodDetails> login() {
+        final AzureTask<AuthMethodDetails> task = new AzureTask<>(null, "Signing in to Azure...", false, () -> {
+            final ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
+            indicator.setIndeterminate(true);
+            return this.doLogin();
+        });
+        return AzureTaskManager.getInstance().runInModal(task).toSingle();
+    }
+
+    private AuthMethodDetails doLogin() {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java:191]: two doLogin methods?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java [hunk 2]
```diff
@@ -272,7 +282,7 @@ private void doSelectCredFilepath() {
             fileDescriptor,
             this.project,
             LocalFileSystem.getInstance().findFileByPath(System.getProperty("user.home"))
-        );
+                                                       );
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java:285]: revert this change

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java [hunk 3]
```diff
@@ -299,19 +309,20 @@ private synchronized BaseADAuthManager doDeviceLogin() {
...
-            } catch (Exception e) {
-                throw new AzureToolkitRuntimeException(e.getMessage(), e);
-            } finally {
-                EventUtil.logEvent(EventType.info, operation, Collections.singletonMap(
-                    AZURE_ENVIRONMENT, CommonSettings.getEnvironment().getName()));
-                operation.complete();
-            }
-        }));
+        final ProgressIndicator progressIndicator = ProgressManager.getInstance().getProgressIndicator();
+        progressIndicator.setText2("Signing in...");
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java:313]:
```  final ProgressIndicator progressIndicator = ProgressManager.getInstance().getProgressIndicator();
            try {	        progressIndicator.setText2("Signing in...");
```
can be wrapped in one util method

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/actions/AzureSignInAction.java [hunk 4]
```diff
@@ -134,23 +136,27 @@ public static void onAzureSignIn(Project project) {
                 });
             }
         } else {
-            doSignIn(authMethodManager, project);
+            doSignIn(authMethodManager, project).subscribe();
         }
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/actions/AzureSignInAction.java:140]: how about moving subscribe() into doSignIn(...) ?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java [hunk 5]
```diff
@@ -342,18 +353,16 @@ private void doCreateServicePrincipal() {
...
-                }
-            }));
+            final ProgressIndicator progressIndicator = ProgressManager.getInstance().getProgressIndicator();
+            progressIndicator.setText2("Loading subscriptions...");
+            try {
+                progressIndicator.setIndeterminate(true);
+                subscriptionManager.getSubscriptionDetails();
+            } catch (Exception ex) {
+                ex.printStackTrace();
+                //LOGGER.error("doCreateServicePrincipal::Task.Modal", ex);
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java:363]: remove commented line? and ex.printStackTrace() is not recommended

Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/ui/SignInWindow.java:363]: remove commented line. and the princkTrace() is not recommended.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/actions/AzureSignInAction.java (hunk 4 above)
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/actions/AzureSignInAction.java:140]: how about moving subscribe() into soSignIn(...) ?

Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/actions/AzureSignInAction.java:140]: no, some other callings need to do something in `subscribe`
